### PR TITLE
MAINT: Fixing broken links with linkchecker

### PR DESCRIPTION
**Pull request by sethtroisi**
I ran `make linkcheck` these were a few bad links it listed that I could find new copies of.

### CODE REVIEW

File: doc/source/tutorial/stats.rst [hunk 1]
```diff
@@ -152,10 +152,8 @@ argument:
     >>> norm.rvs(size=3)
     array([-0.35687759,  1.34347647, -0.11710531])   # random
 
-.. TODO: Replace numpy.random HTML with `numpy.random` once NumPy updates doc
-
 Note that drawing random numbers relies on generators from
-`numpy.random <https://www.numpy.org/devdocs/reference/routines.random.html>`__
```
Comment (sethtroisi) [inline doc/source/tutorial/stats.rst:158]: I didn't verify this works but it's used below so I assume I'm safe.

File: scipy/signal/_peak_finding.py [hunk 2]
```diff
@@ -1036,7 +1036,7 @@ def _identify_ridge_lines(matr, max_distances, gap_thresh):
     ----------
     Bioinformatics (2006) 22 (17): 2059-2065.
     :doi:`10.1093/bioinformatics/btl355`
-    http://bioinformatics.oxfordjournals.org/content/22/17/2059.long
+    https://academic.oup.com/bioinformatics/article/22/17/2059/274284
```
Comment (sethtroisi) [inline scipy/signal/_peak_finding.py:1039]: Original link redirects to here not sure if this should be changed at this time (happy to pull this out and include in a later PR if we care)

File: scipy/special/orthogonal.py [hunk 3]
```diff
@@ -68,7 +68,7 @@
    *Fast computation of Gauss quadrature nodes and
    weights on the whole real line*.
    IMA Journal of Numerical Analysis
-   :doi:`10.1093/imanum/drv002`.
+   https://academic.oup.com/imajna/article-abstract/36/1/337/2363694
```
Comment (sethtroisi) [inline scipy/special/orthogonal.py:71]: Another redirect that I'm happy to remove if they don't beyond in this PR

File: scipy/signal/_peak_finding.py (hunk 2 above)
Comment (rgommers) [reply scipy/signal/_peak_finding.py:1039]: Looks fine to me

File: scipy/special/orthogonal.py (hunk 3 above)
Comment (rgommers) [reply scipy/special/orthogonal.py:71]: This seems undesired, DOIs are unique identifiers that are guaranteed to remain stable and can be easily looked up. We should prefer them to fixed links.

Comment (rgommers) [reply scipy/special/orthogonal.py:71]: Can you undo this and the next ones?

If it's broken, there might be an issue with Sphinx support for `:doi:`, but better to wait for a fix.

Comment (ev-br) [reply scipy/special/orthogonal.py:71]: The doi redirect from
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.roots_hermite.html
seems to work fine, so best left as is.

File: scipy/signal/_peak_finding.py (hunk 2 above)
Comment (pv) [inline scipy/signal/_peak_finding.py:1039]: Actually, it's better to remove the https:// link. The DOI link alone is enough, and will go to the same place.

File: scipy/special/orthogonal.py (hunk 3 above)
Comment (sethtroisi) [reply scipy/special/orthogonal.py]: reverted all the changes to DOI's

File: scipy/signal/_peak_finding.py (hunk 2 above)
Comment (sethtroisi) [reply scipy/signal/_peak_finding.py]: Thanks for the advice. I have updated this entry to be in all three places and removed the link (given the DOI link exists)

File: doc/source/tutorial/stats.rst (hunk 1 above)
Comment (rgommers) [reply doc/source/tutorial/stats.rst:158]: Yep that worked.

**Comment by rgommers** [comment]
LGTM now, merged. Thanks @sethtroisi

**Pull request merged**